src/lib/users/clients/users_finish.rb

```diff
@@ -76,21 +80,19 @@ def write_autoinst
       Users.SetExportAll(false)
       saved = Users.Export
       log.info("Users to import: #{saved}")
-
-      # 2. Read users and settings from the installed system
-      # (bsc#965852, bsc#973639, bsc#974220 and bsc#971804)
-      Users.Read
+      ay_config = Y2Users::Config.new
+      # TODO: support for BTRFS home and also what about login defaults?
+      Y2Users::Autoyast::HashReader.new(saved).read_to(ay_config)
```

Comment:
NP: passing a config to `#read_to` is something that I don't like too much. IMHO, it would be better if our readers simply generates a new config:

~~~
reader = Y2Users::Linux::Reader.new
config = reader.read
~~~ 

The reader object could even cache the resulting config, and return a clone in the next calls:

~~~
config = reader.read #=> result is internally cached
config2 = reader.read #=> returns a clone of the cached config
config3 = reader.read(force: true) #=> discards the cached result and reads again
~~~

What do you think?

Reply:
It makes sense for me.

Reply:
Unless we have a use case, I would simply return the new configuration (with no caching at all). I would move the responsibility to hold the config to the caller. But if we decide to cache the result, you can avoid the `force` parameter by just using a new reader.